selene/src/main.rs

```diff
@@ -391,6 +391,55 @@ fn read_file(checker: &Checker<toml::value::Value>, filename: &Path) {
     );
 }
 
+/// Reads a config file and return content and path. None if not found or empty.
+fn read_config_file() -> Option<(String, Option<PathBuf>)> {

```

Comment:
Why the nested option? There shouldn't be a case where a config's contents exist but not its path.

Reply:
Please go ahead if you have a better solution.

Reply:
This doesn’t answer my question. What would it mean if it returns `Some(String, None)`? Did you intend to return `Option<String, PathBuf>` instead?

Reply:
The PR depends on the existing code. This is needed to satisfy the current architecture. Feel free to improve it if you think you can.

Reply:
My bad, this was actually fixable. Solved in https://github.com/Kampfkarren/selene/pull/591/commits/ade463c78eca904e2d3fd801c3f6a792f258edc6